### PR TITLE
feat: Return 422 on user code errors

### PR DESCRIPTION
**Pull request by vbustamante**
To ensure bad user code is not reported as server errors, we return 422 Unprocessable Content when that's the reason for an API call failure.

To enable this, I had to make the language server errors static and parseable into a rust enum, that has an extra UserCodeException variant containing a string so easily support JS errors on the frontend.

This PR also cleans up code error messages in the frontend

image

### CODE REVIEW

File: lib/cyclone-core/src/progress.rs
Comment (vbustamante) [inline lib/cyclone-core/src/progress.rs:160]: This was the scariest part to add, since it's carried all the way from cyclone back to the DAL basically. From what I've tested it does not break any protocols but I'd appreciate extra eyes

Comment (nickgerace) [reply lib/cyclone-core/src/progress.rs:160]: Looks good to me. You also don't have to worry about #4438 because we only dropped `Reconciliation` funcs from veritech (still "above" in dal, still "below" in cyclone and lang-js).

File: lib/veritech-server/src/server.rs
Comment (nickgerace) [inline lib/veritech-server/src/server.rs:115]: As a heads up, `Cancelled` becomes `Killed` in #4438.

**Review by nickgerace — APPROVED** [review]

File: lib/veritech-server/src/server.rs
Comment (vbustamante) [reply lib/veritech-server/src/server.rs:115]: Oh ok, I'll wait for that to merge before merging this one

**Pull request merged**
